6185 - 4.17 release notes (#6217)
IQSS/dataverse: doc/release-notes/4.17-release-notes.md

This is optional, but all future changes will go to these files. It might be a good idea to reconfigure Solr now or be aware to look for changes to these files in the future, too. Here's how:

1. You will need to replace or modify your `schema.xml` with the recent one (containing XML includes)
2. Copy `schema_dv_mdb_fields.xml` and `schema_dv_mdb_copies.xml` to the same location as the `schema.xml`
3. A re-index is not necessary as long no other changes happened, as this is only a reorganization of Solr fields from a single schema.xml file into multiple files.

In case you use custom metadata blocks, you might find the new `updateSchemaMDB.sh` script beneficial. Again,
see http://guides.dataverse.org/en/4.17/admin/metadatacustomization.html#updating-the-solr-schema

### New Database Settings

[:LoginSessionTimeout](http://guides.dataverse.org/en/4.17/installation/config.html#loginsessiontimeout) controls the session timeout (in minutes) for logged-in users.

## Notes for Tool Developers and Integrators

### Dataset Level Explore Tools

> Tool developers can now define the scope of an external tool as file or dataset. This will allow tools to appear for selection at the dataset or file level. For more information, check out the Guides section on <a href=http://guides.dataverse.org/en/develop/api/external-tools.html/>External Tools</a>. 

Reviewer: ```suggestion
Tool developers can now define the scope of an external tool as file or dataset. This will allow tools to appear for selection at the dataset or file level. For more information, check out the Guides section on <a href=http://guides.dataverse.org/en/4.17/api/external-tools.html/>External Tools</a>.
```

# Dataverse 4.17

This release brings new features, enhancements, and bug fixes to Dataverse. Thank you to all of the community members who contributed code, suggestions, bug reports, and other assistance across the project.

## Release Highlights

### Dataset Level Explore Tools

Tools that integrate with Dataverse can now be launched from the dataset page! This makes it possible to develop and add tools that work across the entire dataset instead of single files. Tools to verify reproducibility and allow researchers to compute on an entire dataset will take advantage of this new infrastructure.

### Performance Enhancements

> Dataverse now allows installation administrators to configure the session timeout for users. Setting this lower will release system resources as configured and will result in better performance throughout a dataverse installation. 

Reviewer: ```suggestion
Dataverse now allows installation administrators to configure the session timeout for logged in users using the new `:LoginSessionTimeout` setting. (Session length for anonymous users has been reduced from 24 hours to 10 minutes.) Setting this lower will release system resources as configured and will result in better performance (less memory use) throughout a Dataverse installation.
```

## Major Use Cases

Newly-supported use cases in this release include:

> - As a user, I can launch and utilize tools that allow me to work across the code, data, and other files in a dataset. 

Reviewer: ```suggestion
 - As a user, I can launch and utilize external tools that allow me to work across the code, data, and other files in a dataset. 
```

- As a developer, I can build external tools to verify reproducibility or allow computation.
- As an installation administrator, I can make a quick configuration change to provide a better experience for my installation's users.

## Notes for Dataverse Installation Administrators

### Configurable User Session Timeout

Idle session timeout for logged-in users has been made configurable in this release. 
The default is now set to 8 hours (this is a change from the previous default value of 24 hours). 
If you want to change it, set the setting :LoginSessionTimeout to the new value *in minutes*. 

> For example, to reduce the timeout to 4 hours:                                                                                                                                                                          

Reviewer: If you add a newline here and indent the curl command to four characters I think it will look better.

Author: will do

### Flexible Solr Schema, optionally reconfigure Solr

With this release, we moved all fields in Solr search index that relate to the default metadata schemas from `schema.xml` to separate files. Custom metadata block configuration of the search index can be more easily automated that way. For details, see admin/metadatacustomization.html#updating-the-solr-schema.

This is optional, but all future changes will go to these files. It might be a good idea to reconfigure Solr now or be aware to look for changes to these files in the future, too. Here's how:

1. You will need to replace or modify your `schema.xml` with the recent one (containing XML includes)
2. Copy `schema_dv_mdb_fields.xml` and `schema_dv_mdb_copies.xml` to the same location as the `schema.xml`
3. A re-index is not necessary as long no other changes happened, as this is only a reorganization of Solr fields from a single schema.xml file into multiple files.

In case you use custom metadata blocks, you might find the new `updateSchemaMDB.sh` script beneficial. Again,

> see admin/metadatacustomization.html#updating-the-solr-schema.

Reviewer: ```suggestion
see http://guides.dataverse.org/en/4.17/admin/metadatacustomization.html#updating-the-solr-schema
```

### Update Scope of Tools

> If you have any external tools enabled on your Dataverse installation, you need to update the manifest.json files to indicate the appropriate scope of the tool

Reviewer: This is something tool authors will need to do, not installation administrators. A database migration script will automatically convert existing tools to file level tools.

Author: got it, will move

Author: @pdurbin - I removed this text in https://github.com/IQSS/dataverse/pull/6217/commits/05300cb968c93728c8cb9512d23657cc0a954c05 because it seems the note below about passing scope is sufficient. Thoughts?

Reviewer: Yes the "Tool developers can now..." stuff below is great. Resolving.

`"scope": "file",`

### New Database Settings

> User Timeout Setting?

Reviewer: ```suggestion
[:LoginSessionTimeout](http://guides.dataverse.org/en/4.17/installation/config.html#loginsessiontimeout) controls the session timeout (in minutes) for logged-in users.
```

## Notes for Tool Developers and Integrators

### Dataset Level Explore Tools

> Tool developers can now define the scope of an external tool as file or dataset. This will allow tools to appear for selection at the dataset or file level. For more information, check out the Guides section on <a href=http://guides.dataverse.org/en/4.17/installation/external-tools.html/>External Tools</a>.

Reviewer: http://guides.dataverse.org/en/develop/api/external-tools.html would be a better page to link to.
